Common/src/config_structure.cpp

```diff
@@ -4555,6 +4551,80 @@ void CConfig::SetPostprocessing(unsigned short val_software, unsigned short val_
     RampOutletPressure = false;
     RampRotatingFrame = false;
   }
+  
+  /* 2nd-order MUSCL is not possible for the continuous adjoint
+   turbulence model. */
+  
+  if (MUSCL_AdjTurb) {
+    SU2_MPI::Error(string("MUSCL_ADJTURB= YES not currently supported.\n") +
+                   string("Please select MUSCL_ADJTURB= NO (first-order)."),
+                   CURRENT_FUNCTION);
+  }
+  
+  /* Check for whether we need a second gradient method to calculate
+   gradients for uwpind reconstruction. Set additional booleans to
+   minimize overhead as appropriate. */
+  
+  if (MUSCL_Flow || MUSCL_Turb || MUSCL_Heat || MUSCL_AdjFlow) {
+    
+    ReconstructionGradientRequired = true;
+    
+    if ((Kind_Gradient_Method_Recon == NO_GRADIENT) ||
+        (Kind_Gradient_Method_Recon == Kind_Gradient_Method)) {
+
+      /* The default behavior if no reconstruction gradient is specified
+       is to use the same gradient as needed for the viscous/source terms
+       without recomputation. If they are using the same method, then
+       we also want to avoid recomputation. */
+
+      ReconstructionGradientRequired = false;
+      Kind_Gradient_Method_Recon     = Kind_Gradient_Method;
+    }
+    
+  }
+  
+  /* Simpler boolean to control allocation of least-squares memory. */
+  
+  if ((Kind_Gradient_Method_Recon == LEAST_SQUARES) ||
+      (Kind_Gradient_Method_Recon == WEIGHTED_LEAST_SQUARES) ||
+      (Kind_Gradient_Method       == LEAST_SQUARES) ||
```

Comment:
Will leave for now. I wrote it so that it would work correctly for LEAST_SQUARES in case somebody really wants to use it in the future. They would only need to remove that error.